src/XMakeBuildEngine/Definition/ToolsetReader.cs

```diff
@@ -153,23 +153,26 @@ ToolsetDefinitionLocations locations
             string overrideTasksPathFromRegistry = null;
             string defaultOverrideToolsVersionFromRegistry = null;
 
+#if FEATURE_WIN32_REGISTRY
             if ((locations & ToolsetDefinitionLocations.Registry) == ToolsetDefinitionLocations.Registry)
             {
-#if FEATURE_WIN32_REGISTRY
-                if (NativeMethodsShared.IsWindows || registryReader != null)
-                // If we haven't been provided a registry reader (i.e. unit tests), create one
-                registryReader = registryReader ?? new ToolsetRegistryReader(environmentProperties, globalProperties);
-
-                // We do not accumulate properties when reading them from the registry, because the order
-                // in which values are returned to us is essentially random: so we disallow one property
-                // in the registry to refer to another also in the registry
-                defaultToolsVersionFromRegistry = registryReader.ReadToolsets(toolsets, globalProperties,
-                    initialProperties, false /* do not accumulate properties */, out overrideTasksPathFromRegistry,
-                    out defaultOverrideToolsVersionFromRegistry);
-            }
+
+                if (NativeMethodsShared.IsWindows || registryReader != null) 
```

Comment:
Ah, my comment was incorrect. `FEATURE_WIN32_REGISTRY` will be enabled even for mono builds, which can run on non-windows platforms too. So, the runtime check `IsWindows` is useful here.